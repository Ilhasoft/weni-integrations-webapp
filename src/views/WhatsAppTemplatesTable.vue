<template>
  <div class="whatsapp-templates">
    <unnnic-breadcrumb :crumbs="crumbs" @crumbClick="handleCrumbClick"></unnnic-breadcrumb>
    <TemplatesHeader />
    <TemplatesTable />
  </div>
</template>

<script>
  import TemplatesHeader from '@/components/WhatsAppTemplates/Header';
  import TemplatesTable from '@/components/WhatsAppTemplates/Table';

  export default {
    name: 'WhatsAppTemplatesTable',
    components: {
      TemplatesHeader,
      TemplatesTable,
    },
    data() {
      return {
        crumbs: [{ name: 'Profile', path: '/apps/my' }, { name: 'Manage templates' }],
      };
    },
    methods: {
      handleCrumbClick(crumb) {
        if (crumb.path) {
          this.$router.push(crumb.path);
        }
      },
    },
  };
</script>

<style lang="scss" scoped></style>
