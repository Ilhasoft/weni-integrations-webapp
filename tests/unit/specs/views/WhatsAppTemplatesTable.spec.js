import { shallowMount, createLocalVue } from '@vue/test-utils';
import WhatsAppTemplatesTable from '@/views/WhatsAppTemplatesTable.vue';

const localVue = createLocalVue();

describe('views/WhatsAppTemplatesTable.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(WhatsAppTemplatesTable, {
      localVue,
      stubs: {
        UnnnicBreadcrumb: true,
        TemplatesHeader: true,
        TemplatesTable: true,
      },
      mocks: {
        $router: {
          push: jest.fn(),
        },
      },
    });
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleCrumbClick()', () => {
    it('should call router.push with desired path if defined', () => {
      const spy = spyOn(wrapper.vm.$router, 'push');
      expect(spy).not.toHaveBeenCalled();
      wrapper.vm.handleCrumbClick({ path: '/path' });
      expect(spy).toHaveBeenCalledTimes(1);
      expect(spy).toHaveBeenCalledWith('/path');
    });

    it('should not call router.push if path is not defined', () => {
      const spy = spyOn(wrapper.vm.$router, 'push');
      expect(spy).not.toHaveBeenCalled();
      wrapper.vm.handleCrumbClick({});
      expect(spy).not.toHaveBeenCalled();
    });
  });
});
