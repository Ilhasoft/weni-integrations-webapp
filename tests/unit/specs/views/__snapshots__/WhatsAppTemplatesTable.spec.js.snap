// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/WhatsAppTemplatesTable.vue should be rendered properly 1`] = `
<div class="whatsapp-templates">
  <unnnicbreadcrumb-stub crumbs="[object Object],[object Object]"></unnnicbreadcrumb-stub>
  <templatesheader-stub title="WhatsApp"></templatesheader-stub>
  <templatestable-stub></templatestable-stub>
</div>
`;
